### ISSUE TITLE
How do I change loss function to CTC??

### ISSUE THREAD

**Issue by rlawoqls7** [georgesterpu/Sigmedia-AVSR#13]
Hi, @georgesterpu.

I also do audio only test in my dataset with LeeYongHyeok.

In this work, I would like to change the loss function from the default loss to CTC loss. 

So I changed line 398 from seq2seq.sequence_loss to CTC loss in the avsr/decoder_unimodal.py, and the loss became inf

I'd appreciate it if you could tell me How do I change loss function to CTC

**Comment by georgesterpu** [comment]
Hello @rlawoqls7 
I am not sure if using the CTC loss on the decoder outputs could work well. 

We previously used the CTC loss with the encoder, complementing the decoder's cross-entropy loss, as on [this diagram](https://georgesterpu.github.io/pics/seq2seq2.png). It was a different task though, lipreading.

To add the CTC loss, you could first project the encoder outputs to a dimension equal to the vocabulary size plus one, and call tf.nn.ctc_loss with the projected encoder outputs as your inputs. Additionally, you may have to convert the labels tensor from the current format (a sequence of integers) to the tf.SparseTensor representation.

**Comment by georgesterpu** [comment]
Please let us know in case you are still facing difficulties with the implementation.
Christian @saamc may be able to assist, as he has recently explored some ideas based on the CTC loss.
